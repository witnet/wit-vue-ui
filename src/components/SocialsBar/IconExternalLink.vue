<template>
  <a :aria-label="url" class="link" :href="url" target="_blank">
    <slot></slot>
    <!-- <icon :class="`icon h-auto ${width} p-[16px] -m-[16px]`" /> -->
  </a>
</template>
<script setup lang="ts">
defineProps({
  url: {
    type: String,
    required: true
  },
  width: {
    type: String,
    default: 'w-[50px]'
  }
  //   icon: {
  //     type: Object,
  //     required: true,
  //   },
})
</script>

<style lang="scss">
// .icon {
//   .social-icon {
//     @apply fill-white-400 transition-all;
//   }
// }
// .icon:hover > .social-icon {
//   @apply fill-white-500;
// }
</style>
