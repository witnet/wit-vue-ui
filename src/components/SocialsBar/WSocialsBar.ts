export const socialsUrls: Record<string, string> = {
  discord: 'https://discord.gg/witnet',
  github: 'https://github.com/witnet',
  telegram: 'https://t.me/witnetio',
  x: 'https://twitter.com/witnet_io'
}
