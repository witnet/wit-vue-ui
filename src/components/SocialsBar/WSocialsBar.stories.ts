import type { Meta, StoryObj } from '@storybook/vue3'
import WSocialsBar from './WSocialsBar.vue'

const meta = {
  title: 'Example/WSocialsBar',
  component: WSocialsBar,
  tags: ['autodocs'],
  argTypes: {},
  args: {}
} satisfies Meta<typeof WSocialsBar>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  render: (args) => ({
    components: { WSocialsBar },
    setup() {
      return { args }
    },
    template: `<WSocialsBar v-bind="args" />`
  }),
  args: {}
}
