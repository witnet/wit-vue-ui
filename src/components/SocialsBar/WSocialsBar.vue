<template>
  <div
    ref="socials"
    class="opacity-60 hover:opacity-100 grid grid-rows-1 grid-flow-row content-center gap-y-md w-fit justify-items-center"
  >
    <IconExternalLink :url="socialsUrls.x">
      <IconTwitter class="icon h-auto w-md" />
    </IconExternalLink>
    <IconExternalLink :url="socialsUrls.telegram">
      <IconTelegram class="icon h-auto w-md" />
    </IconExternalLink>
    <IconExternalLink :url="socialsUrls.discord">
      <IconDiscord class="icon h-auto w-md" />
    </IconExternalLink>
    <IconExternalLink :url="socialsUrls.github">
      <IconGithub class="icon h-auto w-md" />
    </IconExternalLink>

    <!-- TODO(#413): set width to 40px to avoid wrong alignment -->
    <div class="flex mt-3xl w-xl">
      <p class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle">
        Enjoy Smart, Neutral,
      </p>
      <p class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle">
        and Secure Data
      </p>
    </div>
  </div>
</template>
<script setup lang="ts">
// import { ref } from 'vue'

import IconTwitter from '../icons/IconTwitter.vue'
import IconTelegram from '../icons/IconTelegram.vue'
import IconDiscord from '../icons/IconDiscord.vue'
import IconGithub from '../icons/IconGithub.vue'
import IconExternalLink from './IconExternalLink.vue'
import { socialsUrls } from './WSocialsBar'

// const socials = ref(null)
// const { gsap } = useGsap()
// onMounted(() => {
//   startAnimation()
// })
// function startAnimation() {
//   gsap.from(socials.value, {
//     yPercent: 100,
//     ease: 'power2.out',
//     opacity: 0,
//     duration: 1,
//   })
// }
</script>
