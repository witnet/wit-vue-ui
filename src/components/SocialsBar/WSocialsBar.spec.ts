import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import SocialsBar from './WSocialsBar.vue'

describe('SocialsBar', () => {
  it('renders properly', () => {
    const wrapper = mount(SocialsBar)

    expect(wrapper.element).toMatchSnapshot()
  })
})
