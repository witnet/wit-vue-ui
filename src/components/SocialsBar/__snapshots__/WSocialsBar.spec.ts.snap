// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialsBar > renders properly 1`] = `
<div
  class="opacity-60 hover:opacity-100 grid grid-rows-1 grid-flow-row content-center gap-y-md w-fit justify-items-center"
>
  <a
    aria-label="https://twitter.com/witnet_io"
    class="link"
    href="https://twitter.com/witnet_io"
    target="_blank"
  >
    
    <svg
      class="icon h-auto w-md"
      fill="none"
      height="396"
      viewBox="0 0 396 396"
      width="396"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        d="M301.026 37.125H355.608L236.362 173.415L376.645 358.875H266.805L180.774 246.395L82.335 358.875H27.72L155.265 213.098L20.691 37.125H133.32L211.084 139.937L301.026 37.125ZM281.869 326.205H312.114L116.886 68.079H84.4305L281.869 326.205Z"
        fill="#CCCCCC"
      />
    </svg>
    
    <!-- &lt;icon :class="\`icon h-auto \${width} p-[16px] -m-[16px]\`" /&gt; -->
  </a>
  <a
    aria-label="https://t.me/witnetio"
    class="link"
    href="https://t.me/witnetio"
    target="_blank"
  >
    
    <svg
      class="icon h-auto w-md"
      fill="none"
      height="20"
      viewBox="0 0 24 20"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M18.3839 19.779C18.7059 20.007 19.1209 20.064 19.4909 19.924C19.8609 19.783 20.1329 19.467 20.2149 19.084C21.0839 15 23.1919 4.66303 23.9829 0.948026C24.0429 0.668026 23.9429 0.377026 23.7229 0.190026C23.5029 0.00302622 23.1979 -0.0509739 22.9259 0.0500261C18.7329 1.60203 5.81991 6.44703 0.54191 8.40003C0.20691 8.52403 -0.0110904 8.84603 -9.04275e-05 9.19903C0.0119096 9.55303 0.24991 9.86003 0.59291 9.96303C2.95991 10.671 6.06691 11.656 6.06691 11.656C6.06691 11.656 7.51891 16.041 8.27591 18.271C8.37091 18.551 8.58991 18.771 8.87891 18.847C9.16691 18.922 9.47491 18.843 9.68991 18.64C10.9059 17.492 12.7859 15.717 12.7859 15.717C12.7859 15.717 16.3579 18.336 18.3839 19.779ZM7.37391 11.102L9.05291 16.64L9.42591 13.133C9.42591 13.133 15.9129 7.28203 19.6109 3.94703C19.7189 3.84903 19.7339 3.68503 19.6439 3.57003C19.5549 3.45503 19.3909 3.42803 19.2679 3.50603C14.9819 6.24303 7.37391 11.102 7.37391 11.102Z"
        fill="#CCCCCC"
        fill-rule="evenodd"
      />
    </svg>
    
    <!-- &lt;icon :class="\`icon h-auto \${width} p-[16px] -m-[16px]\`" /&gt; -->
  </a>
  <a
    aria-label="https://discord.gg/witnet"
    class="link"
    href="https://discord.gg/witnet"
    target="_blank"
  >
    
    <svg
      class="icon h-auto w-md"
      fill="none"
      height="24"
      viewBox="0 0 21 24"
      width="21"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M18.54 0C19.896 0 21 1.104 21 2.472V24L18.42 21.72L16.968 20.376L15.432 18.948L16.068 21.168H2.46C1.104 21.168 0 20.064 0 18.696V2.472C0 1.104 1.104 0 2.46 0L18.54 0ZM13.908 15.672C16.56 15.588 17.58 13.848 17.58 13.848C17.58 9.984 15.852 6.852 15.852 6.852C14.124 5.556 12.48 5.592 12.48 5.592L12.312 5.784C14.352 6.408 15.3 7.308 15.3 7.308C14.052 6.624 12.828 6.288 11.688 6.156C10.824 6.06 9.996 6.084 9.264 6.18L9.06 6.204C8.64 6.24 7.62 6.396 6.336 6.96C5.892 7.164 5.628 7.308 5.628 7.308C5.628 7.308 6.624 6.36 8.784 5.736L8.664 5.592C8.664 5.592 7.02 5.556 5.292 6.852C5.292 6.852 3.564 9.984 3.564 13.848C3.564 13.848 4.572 15.588 7.224 15.672C7.224 15.672 7.668 15.132 8.028 14.676C6.504 14.22 5.928 13.26 5.928 13.26L6.264 13.464L6.312 13.5L6.359 13.527L6.373 13.533L6.42 13.56C6.72 13.728 7.02 13.86 7.296 13.968C7.788 14.16 8.376 14.352 9.06 14.484C9.96 14.652 11.016 14.712 12.168 14.496C12.732 14.4 13.308 14.232 13.908 13.98C14.328 13.824 14.796 13.596 15.288 13.272C15.288 13.272 14.688 14.256 13.116 14.7C13.476 15.156 13.908 15.672 13.908 15.672ZM8.328 10.068C7.644 10.068 7.104 10.668 7.104 11.4C7.104 12.132 7.656 12.732 8.328 12.732C9.012 12.732 9.552 12.132 9.552 11.4C9.564 10.668 9.012 10.068 8.328 10.068ZM12.708 10.068C12.024 10.068 11.484 10.668 11.484 11.4C11.484 12.132 12.036 12.732 12.708 12.732C13.392 12.732 13.932 12.132 13.932 11.4C13.932 10.668 13.392 10.068 12.708 10.068Z"
        fill="#ccc"
        fill-rule="evenodd"
      />
    </svg>
    
    <!-- &lt;icon :class="\`icon h-auto \${width} p-[16px] -m-[16px]\`" /&gt; -->
  </a>
  <a
    aria-label="https://github.com/witnet"
    class="link"
    href="https://github.com/witnet"
    target="_blank"
  >
    
    <svg
      class="icon h-auto w-md"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M16 0C7.16 0 0 7.16 0 16C0 23.08 4.58 29.06 10.94 31.18C11.74 31.32 12.04 30.84 12.04 30.42C12.04 30.04 12.02 28.78 12.02 27.44C8 28.18 6.96 26.46 6.64 25.56C6.46 25.1 5.68 23.68 5 23.3C4.44 23 3.64 22.26 4.98 22.24C6.24 22.22 7.14 23.4 7.44 23.88C8.88 26.3 11.18 25.62 12.1 25.2C12.24 24.16 12.66 23.46 13.12 23.06C9.56 22.66 5.84 21.28 5.84 15.16C5.84 13.42 6.46 11.98 7.48 10.86C7.32 10.46 6.76 8.82 7.64 6.62C7.64 6.62 8.98 6.2 12.04 8.26C13.32 7.9 14.68 7.72 16.04 7.72C17.4 7.72 18.76 7.9 20.04 8.26C23.1 6.18 24.44 6.62 24.44 6.62C25.32 8.82 24.76 10.46 24.6 10.86C25.62 11.98 26.24 13.4 26.24 15.16C26.24 21.3 22.5 22.66 18.94 23.06C19.52 23.56 20.02 24.52 20.02 26.02C20.02 28.16 20 29.88 20 30.42C20 30.84 20.3 31.34 21.1 31.18C27.42 29.06 32 23.06 32 16C32 7.16 24.84 0 16 0V0Z"
        fill="#ccc"
        fill-rule="evenodd"
      />
    </svg>
    
    <!-- &lt;icon :class="\`icon h-auto \${width} p-[16px] -m-[16px]\`" /&gt; -->
  </a>
  <!-- TODO(#413): set width to 40px to avoid wrong alignment -->
  <div
    class="flex mt-3xl w-xl"
  >
    <p
      class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle"
    >
       Enjoy Smart, Neutral, 
    </p>
    <p
      class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle"
    >
       and Secure Data 
    </p>
  </div>
</div>
`;
